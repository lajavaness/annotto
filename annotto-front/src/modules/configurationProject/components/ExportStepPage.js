import { Checkbox, Form, Radio, Space } from 'antd'
import { isEmpty } from 'lodash'
import { useDispatch } from 'react-redux'
import { useTranslation } from 'react-i18next'
import React, { useCallback } from 'react'

import { ALLITEMS, ANNOTATIONSANDCOMMENTS, CONFIGURATION, HISTORY } from 'shared/enums/exportTypes'

import { exportConfig } from 'modules/configurationProject/actions/configurationProjectActions'

import * as Styled from 'modules/configurationProject/components/__styles__/ExportStepPage.styles'

const ExportStepPage = () => {
	const dispatch = useDispatch()

	const { t } = useTranslation('configurationProject')

	const checkboxOptions = [
		{ label: t('configurationProject:export.configuration'), value: [CONFIGURATION] },
		{ label: t('configurationProject:export.allItems'), value: [ALLITEMS] },
	]

	const radioOptions = [
		{ label: t('configurationProject:export.annotationsAndCommentsHistory'), value: [ANNOTATIONSANDCOMMENTS, HISTORY] },
		{ label: t('configurationProject:export.annotationsAndComment'), value: [ANNOTATIONSANDCOMMENTS] },
	]

	const _onFinish = useCallback(
		({ checked, selected }) => {
			let options = {}
			let types = []

			if (!isEmpty(checked)) {
				types = checked.reduce((acc, result) => [...result, ...acc], [])
			}

			if (!isEmpty(selected)) {
				types = [...types, ...selected]
			}

			if (!isEmpty(types)) {
				types.forEach((type) => (options = { ...options, [type]: true }))
				dispatch(exportConfig(options))
			}
		},
		[dispatch]
	)

	return (
		<Styled.Root>
			<Styled.Form layout="vertical" onFinish={_onFinish}>
				<Space direction="vertical">
					<Form.Item name="checked">
						<Checkbox.Group>
							<Space direction="vertical">
								{checkboxOptions.map(({ label, value }, index) => (
									<Checkbox key={`${value}_${index}`} value={value}>
										{label}
									</Checkbox>
								))}
							</Space>
						</Checkbox.Group>
					</Form.Item>
					<Form.Item name="selected">
						<Radio.Group>
							<Space direction="vertical">
								{radioOptions.map(({ label, value }, index) => (
									<Radio key={`${value}_${index}`} value={value}>
										{label}
									</Radio>
								))}
							</Space>
						</Radio.Group>
					</Form.Item>
					<Form.Item>
						<Styled.Button type="ghost" htmlType="submit">
							{t('configurationProject:export.exportButton')}
						</Styled.Button>
					</Form.Item>
				</Space>
			</Styled.Form>
		</Styled.Root>
	)
}

export default ExportStepPage
