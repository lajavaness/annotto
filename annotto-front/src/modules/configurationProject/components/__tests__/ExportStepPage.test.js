import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { createStore } from 'redux-dynamic-modules'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import ExportStepPage from 'modules/configurationProject/components/ExportStepPage'

const initialState = {}
const getStore = (state) => createStore({ initialState: state })
const getInstance = (state = initialState) => (
	<Provider store={getStore(state)}>
		<ThemeProvider theme={theme}>
			<ExportStepPage />
		</ThemeProvider>
	</Provider>
)

describe('ExportStepPage', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
