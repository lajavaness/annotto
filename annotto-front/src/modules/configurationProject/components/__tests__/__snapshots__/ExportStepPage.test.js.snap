// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportStepPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsBqHK eCiNBh"
  >
    <form
      class="ant-form ant-form-vertical sc-aKXVD hznpLU"
    >
      <div
        class="ant-space ant-space-vertical"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 8px;"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-checkbox-group"
                    id="checked"
                  >
                    <div
                      class="ant-space ant-space-vertical"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 8px;"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value="config"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                          <span>
                            configurationProject:export.configuration
                          </span>
                        </label>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value="allItems"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                          <span>
                            configurationProject:export.allItems
                          </span>
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-bottom: 8px;"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-radio-group ant-radio-group-outline"
                    id="selected"
                  >
                    <div
                      class="ant-space ant-space-vertical"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 8px;"
                      >
                        <label
                          class="ant-radio-wrapper"
                        >
                          <span
                            class="ant-radio"
                          >
                            <input
                              class="ant-radio-input"
                              type="radio"
                              value="annotationsAndComments,withHistory"
                            />
                            <span
                              class="ant-radio-inner"
                            />
                          </span>
                          <span>
                            configurationProject:export.annotationsAndCommentsHistory
                          </span>
                        </label>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <label
                          class="ant-radio-wrapper"
                        >
                          <span
                            class="ant-radio"
                          >
                            <input
                              class="ant-radio-input"
                              type="radio"
                              value="annotationsAndComments"
                            />
                            <span
                              class="ant-radio-inner"
                            />
                          </span>
                          <span>
                            configurationProject:export.annotationsAndComment
                          </span>
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-ghost sc-kmATbt cfrERw"
                    type="submit"
                  >
                    <span>
                      configurationProject:export.exportButton
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
